src/controlobjectscript.cpp

```diff
@@ -0,0 +1,45 @@
+#include <QApplication>
+#include <QtDebug>
+
+#include "controlobjectscript.h"
+
+ControlObjectScript::ControlObjectScript(const ConfigKey& key, QObject* pParent)
+        : ControlObjectSlave(key, pParent) {
+}
+
+ControlObjectScript::~ControlObjectScript() {
+    //qDebug() << "ControlObjectScript::~ControlObjectSlave()";
+}
+
+bool ControlObjectScript::connectScriptFunction(
+        const ControllerEngineConnection& conn) {
+    m_connectedScriptFunction.append(conn);
+    connect(m_pControl.data(), SIGNAL(valueChanged(double, QObject*)),
+            this, SLOT(slotValueChanged(double,QObject*)),
+            static_cast<Qt::ConnectionType>(Qt::QueuedConnection |
+                                            Qt::UniqueConnection));
+    connect(this, SIGNAL(trigger(double, QObject*)),
+            this, SLOT(slotValueChanged(double,QObject*)),
+            static_cast<Qt::ConnectionType>(Qt::QueuedConnection |
+                                            Qt::UniqueConnection));
+    return true;
```

Comment:
This return value does not make any sense. The caller needs to check for failure, but this function will never fail.